### DISCUSSION TITLE
Does Scorecard have a public API or library that can be used to build tools on?

### DISCUSSION THREAD

**Discussion by amaranthjinn** [ossf/scorecard#4353] [General]
I plan to implement a custom dependency analytics tool for my project and want to leverage the scorecard as one of the metrics providers (to check the dependency's ossf score). Is there a recommended scorecard's API library (or libraries) to use to get its utilities, or is the recommended approach to query for the score using http, or calling CLIs from a script-like program?

**Comment by spencerschrock** [comment]
We have a REST API (https://api.scorecard.dev/) for repositories which we have data on, but you may need to collect data yourself for repositories we don't. As Scorecard is written in Go, you can explore what we consider our library entrypoint at https://pkg.go.dev/github.com/ossf/scorecard/v5@v5.0.0/pkg/scorecard#Run.
